patient_portal/messaging/messages.php

```diff
@@ -211,35 +220,40 @@ function getAuthPortalUsers(){
     };
 
     $scope.deleteItem = function (idx) {
-            if( !confirm('Confirm to Delete Selected?') ) return false;
-        var itemToDelete = $scope.pagedItems[$scope.currentPage][idx];
+            if( !confirm($scope.xLate.confirm.one) ) return false;
+            var itemToDelete = $scope.allItems[idx];
         var idxInItems = $scope.items.indexOf(itemToDelete);
-            $scope.deleteMessage(itemToDelete.id);
+            $scope.deleteMessage(itemToDelete.mail_chain); // Just this user's message
         $scope.items.splice(idxInItems,1);
         $scope.search();
+            $scope.init()
 
         return false;
     };
 
         $scope.batchDelete = function ( i ) {
-            if( !confirm('Confirm to delete all selected?') ) return false;
+            if( !confirm($scope.xLate.confirm.all) ) return false;
+            var itemToDelete = [];
             angular.forEach(i, function(o, key) {
                 if(o.hasOwnProperty('deleted')){
-                    var itemToDelete = $scope.items[i.indexOf(o)];
-                    var idxInItems = $scope.items.indexOf(itemToDelete);
-                    $scope.deleteMessage(itemToDelete.id);
-                    $scope.items.splice(idxInItems,1);
+                    itemToDelete.push($scope.items[i.indexOf(o)].id);
                 }
             })
-            location.reload(true);
+            $http.post('handle_note.php', $.param({'task':'massdelete','notejson':JSON.stringify(itemToDelete)}))
+            .success(function(data, status, headers, config) {
+                $window.location.reload();
+            }).error(function(data, status, headers, config) {
+                alert(data);
+            });
+            return false;
         };
 
-        $scope.deleteMessage = function (id){ /* @todo add array for mass delete */
+        $scope.deleteMessage = function (id){
```

Comment:
Was a mass delete implemented?  Sounds like a darn useful feature...

Reply:
Yes you can select messages and delete all